src/search/engines/bing/bing.go

```diff
@@ -60,25 +60,25 @@ func Search(ctx context.Context, query string, relay *bucket.Relay, options engi
 		}
 	})
 
-	localeParam := getLocale(options)
-
-	retErrors := make([]error, options.MaxPages)
-
-	colCtx := colly.NewContext()
-	colCtx.Put("page", strconv.Itoa(1))
-
-	urll := Info.URL + query + localeParam
-	anonUrll := Info.URL + anonymize.String(query) + localeParam
+	retErrors := make([]error, options.Pages.Start+options.Pages.Max)
```

Comment:
If someone asks for page 100 we allocate an array of 100 elements? It might be better to just allocate options.Pages.Max and have an index variable keep track of it.
Also: Even if we do do that^ right now someone can asks for 100000000 pages (the Max variable) and we will just allocate arrays of that size, huh? That might be bad (i.e. lead to a DoS attack).
We may have more spots like this in the code..

Reply:
Oh! Appending to the slice and letting it expand instead of making it that size would solve that problem right?